app/src/main/java/org/groundplatform/android/repository/UserRepository.kt

```diff
@@ -99,4 +99,26 @@ constructor(
       false
     }
   }
+
+  /**
+   * Returns true if the currently logged in user can delete the given LOI. This is allowed if:
+   * - The user is a survey organizer, OR
+   * - The user created the LOI (data collector who created their own site)
+   */
+  suspend fun canDeleteLoi(creatorUserId: String): Boolean {
```

Comment:
Can we pass `LOI` to the param instead of the user id?

This would help encapsulate the logic within the method rather than splitting it between here and the HomeScreenMapContainerViewModel.